emcc.py

```diff
@@ -1641,7 +1641,8 @@ def include_and_export(name):
       exit_with_error('If pthreads and memory growth are enabled, MAXIMUM_MEMORY must be set')
 
     if shared.Settings.EXPORT_ES6 and not shared.Settings.MODULARIZE:
-      exit_with_error('EXPORT_ES6 requires MODULARIZE to be set')
+      # EXPORT_ES6 requires output to be a module
+      shared.Settings.MODULARIZE = 1
```

Comment:
Is there a way we could still throw an error in case user explicitly set `-s MODULARIZE=0` but not when they omitted the option?

Reply:
Great. Let's do that, @algestam.

Reply:
Thanks, will implement this and update the PR.

Reply:
Looks great, thanks! It's odd there isn't an existing test for the previous behaviour, but perhaps worth adding one (to test couple of scenarios - no `MODULARIZE` is provided, explicit `MODULARIZE=1` is provided and explicit `MODULARIZE=0` is provided)?

Reply:
There's no tests for EXPORT_ES6 at all... it's a big blind spot.

Reply:
`EXPORT_ES6` is automatically set when `.mjs` is uses as the output filename so we do have a tiny bit of testing :)

```
$ git grep "\.mjs" tests/*.py
tests/test_other.py:    self.run_process([EMCC, '-o', 'hello_world.mjs', path_from_root('tests', 'hello_world.c')])
tests/test_other.py:    with open('hello_world.mjs') as f:
tests/test_other.py:    # self.assertContained('hello, world!', self.run_js('hello_world.mjs'))
tests/test_other.py:    self.run_process([EMCC, '-o', 'hello_world.mjs', '-pthread', '-O1',
tests/test_other.py:    with open('hello_world.mjs') as f:
```

Reply:
I can look into adding tests that checks that:
* `MODULARIZE` also is set when `EXPORT_ES6` is set
* An error is raised if `EXPORT_ES6` is set and `MODULARIZE` is explicitly disabled

Is `tests/test_other.py` a good place to add these tests?

Reply:
@algestam Yeah, that sounds good. Probably add them somewhere right after existing `mjs` test.

Reply:
Thanks, will give it a try and update the PR.